File: python/word_frequency.py
"""
    create a word frequency list from a text file
"""
def word_frequency(filename):
    """
        create a word frequency list from a text file
    """
    with open(filename, 'r') as f:
        words = f.read().split()
    d = {}
    for word in words:
        if word in d:
            d[word] += 1
        else:
            d[word] = 1
    return d

def main():
    """
        main function
    """
    print(word_frequency('test.txt'))

File: golang/word_frequency.go
// create a map with key as the word and value as the frequency of the word
// read the text file and store the words in the map
// print the map

type WordFrequency struct {
	word string
	freq int
}

// caculate the frequency of the words in the text file with go routines


func main() {
	file, err := os.Open("input.txt")
	if err != nil {
		log.Fatal(err)
	}
	defer file.Close()

	scanner := bufio.NewScanner(file)
	scanner.Split(bufio.ScanWords)

	var words map[string]int
	words = make(map[string]int)

	for scanner.Scan() {
		words[scanner.Text()]++
	}

	if err := scanner.Err(); err != nil {
		log.Fatal(err)
	}

	for k, v := range words {
		fmt.Printf("%s: %d\n", k, v)
	}
}
